flowpipe/graph.py

```diff
@@ -108,62 +108,95 @@ def add_node(self, node):
             LogObserver.push_message(
                 "Node '{0}' is already part of this Graph".format(node.name))
 
-    def evaluate(self, threaded=False, submission_delay=0.1, raise_after=None):
-        """Evaluate all Nodes.
+    def evaluate(self, mode="linear", skip_clean=False,
+                 submission_delay=0.1, raise_after=None):
+        """Evaluate all Nodes in the graph.
+
+        Sorts the nodes in the graph into a resolution order and evaluates the
+        nodes. Nodes that do not depend on each other can be parallelized
+        either by threading or multiprocessing, see the "mode" keyword.
+
+        Note that no checks are in place whether the node execution is actually
+        thread-safe or fit for multiprocessing. It is assumed to be given if
+        the respective mode is selected.
 
         Args:
-            threaded (bool): Whether to execute each node in a separate thread.
+            mode (str): The evaluation mode. Possible modes are
+                * linear : Iterates over all nodes in a single thread
+                * threading : Spawns independent nodes in new threads
+                * multiprocessing : Spawns independent nodes in new processes
```

Comment:
"Spawns a deamonized process for each node"?

Reply:
Is
"Evaluates each node in a new process"
good, too? Then the documentation for the modes fits in one line each ;)

Reply:
sounds good too, let's do that!